Bottom: ce892ed9899555fb59e9db7ed56154782a9b6f6c
Top:    34b9933cf18a91e480cb9f9a2264e6b218c9dfb4
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-07-23 15:13:13 -0500

Rework check for first install

Create a fact for nexus_first_install rather than checking stdout
of a command in several places.



---
diff --git a/roles/nexus3/tasks/main.yml b/roles/nexus3/tasks/main.yml
index 1cfa7111..0e570bf4 100644
--- a/roles/nexus3/tasks/main.yml
+++ b/roles/nexus3/tasks/main.yml
@@ -32,7 +32,7 @@
     - nuget-group
     - nuget-hosted
     - nuget.org-proxy
-  when: nexus_data_dir_contents.stdout == "" and nexus_delete_default_repos
+  when: nexus_first_install and nexus_delete_default_repos
 
 - name: Deleting default blobstore
   include: delete_blobstore_each.yml
@@ -48,7 +48,7 @@
     - name: "{{ nexus_blob_names.mvn.blob }}"
     - name: "{{ nexus_blob_names.gitlfs.blob }}"
     - name: "{{ nexus_blob_names.yum.blob }}"
-  when: nexus_data_dir_contents.stdout == "" and nexus_delete_default_blobstore
+  when: nexus_first_install and nexus_delete_default_blobstore
 
 - include: setup_ldap_each.yml
   with_items: "{{ ldap_connections }}"
diff --git a/roles/nexus3/tasks/nexus_install.yml b/roles/nexus3/tasks/nexus_install.yml
index b3bf57b5..22c68332 100644
--- a/roles/nexus3/tasks/nexus_install.yml
+++ b/roles/nexus3/tasks/nexus_install.yml
@@ -210,6 +210,9 @@
   check_mode: no
   changed_when: false
 
+- set_fact:
+    nexus_first_install: "{{ nexus_data_dir_contents.stdout == '' }}"
+
 - name: Clean cache for upgrade process
   file:
     path: "{{ nexus_data_dir }}/clean_cache"
